File: go/1716.go
func totalMoney(n int) int {

  nums := make([]int, 1)
  nums[0] = 1

  for i:=1; i < n; i++ {
    if i % 7 == 0 {
      nums = append(nums, nums[i-7] + 1)
    } else {
      nums = append(nums, nums[i-1] + 1)
    }
  }

  result := 0
  for _,v := range nums {
    result += v
  }
  return result
}

File: py/1716.py
class Solution:
    def totalMoney(self, n: int) -> int:
        path = [1]

        for i in range(1,n):
            if i % 7 == 0:
                path.append(path[i-7] + 1)
            else:
                path.append(path[i-1] + 1)
        return sum(path)
